downloads/swan-lake-release-notes/swan-lake-2201.3.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,473 @@
+
+### Improvements
+
+#### Semantic API
+Improved the `constValue()` method to retrieve the constant value as an object from the constant symbol
+
+#### Syntax API
+
+Added a few methods to the `NodeParser` API. The methods below have been introduced for the `NodeParser` class.
+  - `Node parseObjectMember(String text)`
+  - `ModulePartNode parseModulePart(String text)`
+  - `IntermediateClauseNode parseIntermediateClause(String text, boolean allowActions)`
+
+### Backward-incompatible changes
+
+Updated the `NodeFactory` methods to allow optional terminating semicolons for module-level declarations. The methods below in the `NodeFactory` have been updated with an extra parameter for the optional semicolon token.
+
+  - `createServiceDeclarationNode`
+  - `createFunctionBodyBlockNode`
+  - `createEnumDeclarationNode`
+  - `createClassDefinitionNode`
+
+### Bug fixes
+
+To view bug fixes, see the [GitHub milestone for 2201.3.0 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.3.0+label%3AType%2FBug+is%3Aclosed+label%3ATeam%2FCompilerFETools+).
+
+## Runtime updates
+
+### Improvements
+
+#### Show the total strand count in strand dump
+
+The total number of strand groups and strands created in a Ballerina program is now shown in the strand dump.
+
+#### Allow ambiguous target types in `fromJsonWithType` and `cloneWithType`
+
+The `fromJsonWithType` and `cloneWithType` functions returned an error in the earlier implementation when the user-specified target type was a union type consisting of multiple convertible types for a given source value. Now, for such user-specified target types, the operation succeeds without returning an error by returning a value whose inherent type is as follows. If the given source value belongs to at least one of the member types of the target union type, the inherent type of the returned value is the first found type to which the source value belongs. Otherwise, the inherent type of the returned value is the first convertible type in the union.
+
+For example, the operations below will not return an error.
+
+```ballerina
+type PetByAge record {
+    int age;
+    string nickname?;
+};
+
+type PetByType record {
+    "Cat"|"Dog" pet_type;
+    boolean hunts?;
+};
+
+type Pet PetByAge|PetByType;
+
+public function main() {
+    json j = {
+        "nickname": "Fido",
+        "pet_type": "Dog",
+        "age": 4
+    };
+    Pet|error pet = j.fromJsonWithType(); // inherent type of pet will be PetByAge
+
+    j = {x: 1};
+    record {float x;}|record {int x;}|error v = j.cloneWithType(); // inherent type of v will be record {int x;}
+}
+```
+
+#### Configurable support for the array of tables
+
+The configurable feature is improved to support the array of tables using a two-dimensional TOML array of in-line tables.
+
+For example, if the configurable variables are defined in the following way,
```

Comment:
Isn't it better to avoid half sentences around the code snippets?

Reply:
@HindujaB @Nadeeshan96 @warunalakshitha

Reply:
@anupama-pathirage - I think it is fine as it brings more clarity. Any suggestions to improve further?

Reply:
Here we need to mention the mapping between the `bal` code and the `toml` file. Either we can have it like this, or we can have the entire sentence first and then the `bal` and `toml` snippets afterward. 
```suggestion
For example, the configurable variables can be configured in the `Config.toml` file as follows.
<ballerina_code>
<toml_content>
```
IMO, having half sentences will be more clear to understand the mapping.